app/src/main/java/com/amaze/filemanager/filesystem/ftpserver/commands/AVBL.kt

```diff
@@ -0,0 +1,119 @@
+package com.amaze.filemanager.filesystem.ftpserver.commands
+
+import android.util.Log
+import com.amaze.filemanager.application.AppConfig
+import com.amaze.filemanager.filesystem.ftpserver.AndroidFileSystemFactory
+import org.apache.ftpserver.command.AbstractCommand
+import org.apache.ftpserver.ftplet.DefaultFtpReply
+import org.apache.ftpserver.ftplet.FtpFile
+import org.apache.ftpserver.ftplet.FtpReply.REPLY_213_FILE_STATUS
+import org.apache.ftpserver.ftplet.FtpReply.REPLY_502_COMMAND_NOT_IMPLEMENTED
+import org.apache.ftpserver.ftplet.FtpReply.REPLY_550_REQUESTED_ACTION_NOT_TAKEN
+import org.apache.ftpserver.ftplet.FtpRequest
+import org.apache.ftpserver.impl.FtpIoSession
+import org.apache.ftpserver.impl.FtpServerContext
+import org.apache.ftpserver.usermanager.impl.WriteRequest
+import java.io.File
+
+/**
+ * Implements FTP extension AVBL command, to answer device remaining space in FTP command.
+ *
+ * Only supports [com.amaze.filemanager.filesystem.ftpserver.RootFileSystemFactory] and
+ * [org.apache.ftpserver.filesystem.nativefs.NativeFileSystemFactory]. Otherwise will simply return
+ * 550 Access Denied.
+ *
+ * See [Draft spec](https://www.ietf.org/archive/id/draft-peterson-streamlined-ftp-command-extensions-10.txt)
+ */
+class AVBL : AbstractCommand() {
+
+    companion object {
+        private const val TAG = "AVBL"
```

Comment:
Please don't use tag anymore to log, use slf4j logger.you can take reference from other classes.